Importing components with alias in monorepo causes VSCode/Volar to fail finding modules/type declarations

### Environment

- Operating System: Darwin
- Node Version:     v18.16.0
- Nuxt Version:     3.6.1
- Nitro Version:    2.5.2
- Package Manager:  pnpm@8.6.6
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://stackblitz.com/github/samydoesit/nuxt-monorepo-import-reproduction
https://github.com/samydoesit/nuxt-monorepo-import-reproduction

### Describe the bug

Hi, i am experiencing issues while importing components in a monorepo using an alias.

The problem arises when using VSCode with the Volar extension. Since our ui directory is located outside of the Nuxt project directory, VSode (Volar) is unable to locate the module or type declaration.

However, if i manually add the path of the alias to the "includes" field in nuxt's tsconfig, the resolution in VSCode works.

### Additional context

_No response_

### Logs

```shell-script
Cannot find module '@ui/components/button/Button.vue' or its corresponding type declarations.
```

--- a/packages/nuxi/src/utils/prepare.ts
+++ b/packages/nuxi/src/utils/prepare.ts
@@ -52,6 +52,7 @@ export const writeTypes = async (nuxt: Nuxt) => {
   const basePath = tsConfig.compilerOptions!.baseUrl ? resolve(nuxt.options.buildDir, tsConfig.compilerOptions!.baseUrl) : nuxt.options.buildDir
 
   tsConfig.compilerOptions = tsConfig.compilerOptions || {}
+  tsConfig.include = tsConfig.include || []
 
   for (const alias in aliases) {
     if (excludedAlias.some(re => re.test(alias))) {
@@ -63,8 +64,17 @@ export const writeTypes = async (nuxt: Nuxt) => {
     if (stats?.isDirectory()) {
       tsConfig.compilerOptions.paths[alias] = [absolutePath]
       tsConfig.compilerOptions.paths[`${alias}/*`] = [`${absolutePath}/*`]
+
+      if (!absolutePath.startsWith(nuxt.options.srcDir)) {
+        tsConfig.include.push(absolutePath)
+        tsConfig.include.push(`${absolutePath}/*`)
+      }
     } else {
       tsConfig.compilerOptions.paths[alias] = [absolutePath.replace(/(?<=\w)\.\w+$/g, '')] /* remove extension */
+
+      if (!absolutePath.startsWith(nuxt.options.srcDir)) {
+        tsConfig.include.push(absolutePath.replace(/(?<=\w)\.\w+$/g, ''))
+      }
     }
   }
 

--- a/packages/nuxi/src/utils/prepare.ts
+++ b/packages/nuxi/src/utils/prepare.ts
@@ -3,10 +3,13 @@ import { isAbsolute, join, relative, resolve } from 'pathe'
 import type { Nuxt, TSReference } from '@nuxt/schema'
 import { defu } from 'defu'
 import type { TSConfig } from 'pkg-types'
+import { withTrailingSlash } from 'ufo'
 import { getModulePaths, getNearestPackage } from './cjs'
 
 export const writeTypes = async (nuxt: Nuxt) => {
   const modulePaths = getModulePaths(nuxt.options.modulesDir)
+
+  const rootDirWithSlash = withTrailingSlash(nuxt.options.rootDir)
 
   const tsConfig: TSConfig = defu(nuxt.options.typescript?.tsConfig, {
     compilerOptions: {
@@ -31,7 +34,7 @@ export const writeTypes = async (nuxt: Nuxt) => {
       join(relative(nuxt.options.buildDir, nuxt.options.rootDir), '**/*'),
       ...nuxt.options.srcDir !== nuxt.options.rootDir ? [join(relative(nuxt.options.buildDir, nuxt.options.srcDir), '**/*')] : [],
       ...nuxt.options._layers.map(layer => layer.config.srcDir ?? layer.cwd)
-        .filter(srcOrCwd => !srcOrCwd.startsWith(nuxt.options.rootDir) || srcOrCwd.includes('node_modules'))
+        .filter(srcOrCwd => !srcOrCwd.startsWith(rootDirWithSlash) || srcOrCwd.includes('node_modules'))
         .map(srcOrCwd => join(relative(nuxt.options.buildDir, srcOrCwd), '**/*')),
       ...nuxt.options.typescript.includeWorkspace && nuxt.options.workspaceDir !== nuxt.options.rootDir ? [join(relative(nuxt.options.buildDir, nuxt.options.workspaceDir), '**/*')] : []
     ],
@@ -65,14 +68,14 @@ export const writeTypes = async (nuxt: Nuxt) => {
       tsConfig.compilerOptions.paths[alias] = [absolutePath]
       tsConfig.compilerOptions.paths[`${alias}/*`] = [`${absolutePath}/*`]
 
-      if (!absolutePath.startsWith(nuxt.options.srcDir)) {
+      if (!absolutePath.startsWith(rootDirWithSlash)) {
         tsConfig.include.push(absolutePath)
         tsConfig.include.push(`${absolutePath}/*`)
       }
     } else {
       tsConfig.compilerOptions.paths[alias] = [absolutePath.replace(/(?<=\w)\.\w+$/g, '')] /* remove extension */
 
-      if (!absolutePath.startsWith(nuxt.options.srcDir)) {
+      if (!absolutePath.startsWith(rootDirWithSlash)) {
         tsConfig.include.push(absolutePath.replace(/(?<=\w)\.\w+$/g, ''))
       }
     }

